utils/hotspot-cli/src/main.rs

```diff
@@ -0,0 +1,438 @@
+        collection_metadata,
+        collection_master_edition,
+        data_only_config,
+        entity_creator,
+        dao,
+        key_to_asset,
+        tree_authority,
+        recipient: program.payer(),
+        merkle_tree: data_only_config_acc.merkle_tree,
+        data_only_escrow,
+        bubblegum_signer,
+        token_metadata_program: token_metadata_pid,
+        log_wrapper: noop_pid,
+        bubblegum_program: BGUM_PID,
+        compression_program: compression_pid,
+        system_program: system_program::id(),
+    }
+}
+
+fn construct_onboard_iot_accounts(
+    program: &Program,
+    hnt_mint: Pubkey,
+    entity_key: &Vec<u8>,
+) -> OnboardDataOnlyIotHotspotV0 {
+    let compression_program = Pubkey::from_str(COMPRESSION).unwrap();
+    let iot_mint = Pubkey::from_str(IOT_MINT).unwrap();
+    let dc_mint = Pubkey::from_str(DC_MINT).unwrap();
+
+    let (dao, _dao_bump) =
+        Pubkey::find_program_address(&["dao".as_bytes(), hnt_mint.as_ref()], &HEM_PID);
+    let (sub_dao, _sd_bump) =
+        Pubkey::find_program_address(&["sub_dao".as_bytes(), iot_mint.as_ref()], &HSD_PID);
+
+    let (rewardable_entity_config, _rec_bump) = Pubkey::find_program_address(
+        &[
+            "rewardable_entity_config".as_bytes(),
+            sub_dao.as_ref(),
+            "IOT".as_bytes(),
+        ],
+        &HEM_PID,
+    );
+
+    let mut hasher = Sha256::new();
+    hasher.update(entity_key);
+    let hash = hasher.finalize();
+
+    let (iot_info, _info_bump) = Pubkey::find_program_address(
+        &[
+            "iot_info".as_bytes(),
+            rewardable_entity_config.key().as_ref(),
+            &hash,
+        ],
+        &HEM_PID,
+    );
+
+    let (data_only_config, _data_only_bump) =
+        Pubkey::find_program_address(&["data_only_config".as_bytes(), dao.as_ref()], &HEM_PID);
+
+    let data_only_config_acc_raw = program.rpc().get_account(&data_only_config).unwrap();
+    let data_only_config_acc =
+        DataOnlyConfigV0::try_deserialize(&mut data_only_config_acc_raw.data.as_slice()).unwrap();
+
+    let (key_to_asset, _kta_bump) =
+        Pubkey::find_program_address(&["key_to_asset".as_bytes(), dao.as_ref(), &hash], &HEM_PID);
+
+    let (dc, _dc_bump) =
+        Pubkey::find_program_address(&["dc".as_bytes(), dc_mint.as_ref()], &DC_PID);
+
+    OnboardDataOnlyIotHotspotV0 {
+        payer: program.payer(),
+        dc_fee_payer: program.payer(),
+        iot_info,
+        hotspot_owner: program.payer(),
+        merkle_tree: data_only_config_acc.merkle_tree,
+        dc_burner: get_associated_token_address(&program.payer(), &dc_mint),
+        rewardable_entity_config,
+        data_only_config,
+        dao,
+        key_to_asset,
+        sub_dao,
+        dc_mint,
+        dc,
+        compression_program,
+        data_credits_program: DC_PID,
+        token_program: Pubkey::from_str("TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA").unwrap(),
```

Comment:
can't get this from a const function somewhere?

Reply:
Done